extensions/charts/verify.go

```diff
@@ -37,3 +37,66 @@ func VerifyChartInstall(client *catalog.Client, chartNamespace, chartName string
 	})
 	return err
 }
+
+// VerifyChartUpgrade verifies that the app from a chart was successfully upgraded
+func VerifyChartUpgrade(client *catalog.Client, chartNamespace, chartName, chartVersion string) error {
+	watchAppInterface, err := client.Apps(chartNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + chartName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*v1.App)
+
+		state := app.Status.Summary.State
+		if state == string(v1.StatusPendingUpgrade) {
+			return true, nil
+		}
+
+		if state == string(v1.StatusFailed) {
+			return false, errors.New("chart upgrade has failed")
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	watchAppInterface, err = client.Apps(chartNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + chartName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*v1.App)
+
+		state := app.Status.Summary.State
+		if state == string(v1.StatusDeployed) {
+			if app.Spec.Chart.Metadata.Version == chartVersion {
+				return true, nil
+			}
+			return false, errors.New("chart not upgraded to expected version")
```

Comment:
Can you change this to show the expected version it tried to upgrade to in a situation that it fails? This would mainly be for debugging purposes.